aten/src/ATen/THLongStorageView.h

```diff
@@ -53,22 +53,30 @@ class THLongStorageView {
         break;
     }
 
-    if(zero_dim_to_one && ref.size() == 0) {
+    if (zero_dim_to_one && ref.size() == 0) {
       // make storage of size 0 actually a 1-length storage with 1 element
       // so that our 0-dim tensors get allocated as 1-dim inside TH
+
       one = 1;
-      storage.data_ptr = {&one, kCPU}; // non-owning
-      storage.size = 1;
+      storage = new StorageImpl(
```

Comment:
I found your leak, it's here. storage is a raw pointer, and you aren't explicitly freeing it, so you're leaking these StorageImpl structs. You should store a `Storage` instead to avoid the leak.

Reply:
(Storing a `Storage` pointery thing here is still suboptimal, because the lifetime of pointers like `&one` is tied to the lifetime of THLongStorageView, so if storage lives on beyond this class you'll still corrupt memory. But I guess since the idea is to get rid of this class entirely I can live with it... as long as there's a comment.)

(I don't know why you stopped storing StorageImpl in this class directly, but I guess you had your reasons?)

Reply:
Thanks! Great find :) - I used a pointer, because the assignment constructor for StorageImpl is deleted and I didn't want to implement it.